examples/cacoh.py

```diff
@@ -0,0 +1,496 @@
+Compute multivariate coherency/coherence
+========================================
+
+This example demonstrates how canonical coherency (CaCoh)
+:footcite:`VidaurreEtAl2019` - a multivariate method based on coherency - can
+be used to compute connectivity between whole sets of sensors, alongside
+spatial patterns of the connectivity.
+"""
+
+# Authors: Mohammad Orabe <orabe.mhd@gmail.com>
+#          Thomas S. Binns <t.s.binns@outlook.com>
+# License: BSD (3-clause)
+
+# %%
+import numpy as np
+from matplotlib import pyplot as plt
+
+import mne
+from mne_connectivity import seed_target_indices, spectral_connectivity_epochs
+
+###############################################################################
+# Background
+# ----------
+#
+# Multivariate forms of signal analysis allow you to simultaneously consider
+# the activity of multiple signals. In the case of connectivity, the
+# interaction between multiple sensors can be analysed at once, producing a
+# single connectivity spectrum. This approach brings not only practical
+# benefits (e.g. easier interpretability of results from the dimensionality
+# reduction), but can also offer methodological improvements (e.g. enhanced
+# signal-to-noise ratio).
+#
+# A popular bivariate measure of connectivity is coherency/coherence, which
+# looks at the correlation between two signals in the frequency domain.
+# However, in cases where interactions between multiple signals are of
+# interest, computing connectivity between all possible combinations of signals
+# leads to a very large number of results which is difficult to interpret. A
+# common approach is to average results across these connections, however this
+# risks reducing the signal-to-noise ratio of results and burying interactions
+# that are present between only a small number of channels.
+#
+# Canonical coherency (CaCoh) is a multivariate form of coherency that uses
+# spatial filters to extract the relevant components of connectivity in a
+# frequency-resolved manner :footcite:`VidaurreEtAl2019`. It is similar to
+# multivariate methods based on the imaginary part of coherency (MIC & MIM
+# :footcite:`EwaldEtAl2012`; see :doc:`mic_mim` and
+# :doc:`compare_coherency_methods`), which are also supported by
+# MNE-Connectivity.
+
+
+###############################################################################
+# Data Simulation
+# ---------------
+#
+# To demonstrate the CaCoh method, will we use some simulated data consisting
+# of an interaction between signals in a given frequency range. Here, we
+# simulate two sets of interactions:
+#
+# - 5 seeds and 3 targets interacting in the 10-12 Hz frequency range.
+# - 5 seeds and 3 targets interacting in the 23-25 Hz frequency range.
+#
+# We can consider the seeds and targets to be signals of different modalities,
+# e.g. cortical EEG signals and subcortical LFP signals, cortical EEG signals
+# and muscular EMG signals, etc.... We use the function below to simulate these
+# signals.

```

Comment:
```suggestion
# "Seeds" and "targets" are keywords indicating the directionality of the connectivity we are
# trying to estimate.
# We can consider the "seeds" and "targets" to be signals of different modalities,
# e.g. cortical EEG signals and subcortical LFP signals, cortical EEG signals
# and muscular EMG signals, etc.... We use the function below to simulate these
# seed and target signals.

```

Is this correct?

Reply:
It is correct, but since CaCoh is an undirected method the assignment to seeds or targets is arbitrary, so I think mentioning "directionality" here could confuse people.